src/lib/onboarding.js

```diff
@@ -14,24 +61,48 @@ const calculateAgeGroup = (birthYear, birthMonth) => {
     return '[17+]';
 };
 
-// add logic when implementing the eligibility check in [UEPR-71]
-const onboardingTestGroup = () => 'Displayed Onboarding Journeys';
+const onboardingTestGroup = user =>
+    (user.id % 2 === 0 ?
+        ONBOARDING_TESTING_GROUP_A_NAME :
+        ONBOARDING_TESTING_GROUP_B_NAME);
+
+export const onboardingEligibilityCheck = (user, permissions) =>
+    Object.keys(user).length !== 0 &&
+    Object.keys(permissions).length !== 0 &&
+    isOnboardingTestingToggledOn() &&
+    isUserEligible(user) &&
+    !isAdmin(permissions) &&
+    !isMuted(permissions) &&
+    !isBanned(user);
 
 export const triggerAnalyticsEvent = eventVaribles => {
     window.dataLayer = window.dataLayer || [];
-    
+
     window.dataLayer.push({
         ...eventVaribles
     });
 };
 
 export const sendUserProperties = user => {
+    if (
+        !isOnboardingTestingToggledOn() ||
+        !isRegisteredInRange(user) ||
+        !isCurrentDayInRange()
+    ) {
+        window.dataLayer.push({
+            testGroup: null,
+            ageGroup: null,
+            gender: null
+        });
```

Comment:
My understanding is that user properties persist unless we pass different value for a particular one. So we have to unset the values for users that are not part of the A/B testing so that they don't interfere with the statistics.

Reply:
Ah yeah, and multiple users could use the same machine. Makes sense